vlog4j-parser/src/main/java/org/semanticweb/vlog4j/parser/javacc/JavaCCParserBase.java

```diff
@@ -222,6 +235,24 @@ void resetVariableSets() {
 		this.headUniVars.clear();
 	}
 
+	/**
+	 * Convert a throwable into a ParseException.
+	 *
+	 * @param message The error message.
+	 * @param cause   The {@link Throwable} that caused this exception.
+	 *
+	 * @return A {@link ParseException} with appropriate cause and message.
+	 */
+	protected ParseException makeParseExceptionWithCause(String message, Throwable cause) {
```

Comment:
it may be an overhead in this case, but it is cleaner to extend ParseException and add the cause in the constructor.

This is ok for now.

Reply:
Okay, will keep this in mind for the Parser refactoring.